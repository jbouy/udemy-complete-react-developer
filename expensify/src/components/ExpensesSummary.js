import React from "react";
import { connect } from "react-redux";
import numeral from "numeral";

import getExpensesTotal from "../selectors/expenses-total";
import selectExpenses from "../selectors/expenses";

export const ExpensesSummary = ({ count, total }) => {
  const expenseWord = count === 1 ? "expense" : "expenses";
  const formattedExpenseTotal = numeral(total / 100).format("$0,0.00");

  return (
    <div>
      <h1>
        Viewing {count} {expenseWord} totalling {formattedExpenseTotal}
      </h1>
    </div>
  );
};

const mapStateToProps = state => {
  const expenses = selectExpenses(state.expenses, state.filters) || [];

  return {
    count: expenses.length,
    total: getExpensesTotal(expenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
