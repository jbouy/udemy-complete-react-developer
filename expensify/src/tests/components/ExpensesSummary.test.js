import React from "react";
import { shallow } from "enzyme";

import { ExpensesSummary } from "../../components/ExpensesSummary";

let wrapper, props;

describe("ExpensesSummary", () => {
  beforeEach(() => {
    props = {
      count: 2,
      total: 23145
    };

    wrapper = shallow(<ExpensesSummary {...props} />);
  });

  test("should render one expense correctly", () => {
    wrapper.setProps({
      count: 1,
      total: 100
    });

    expect(wrapper).toMatchSnapshot();
  });

  test("should render multiple expenses correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
