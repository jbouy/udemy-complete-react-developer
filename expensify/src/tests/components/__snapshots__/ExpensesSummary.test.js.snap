// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpensesSummary should render multiple expenses correctly 1`] = `
<div>
  <h1>
    Viewing 
    2
     
    expenses
     totalling 
    $231.45
  </h1>
</div>
`;

exports[`ExpensesSummary should render one expense correctly 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $1.00
  </h1>
</div>
`;
